Improve vendor-boringssl.sh script to make it work better

xtremekforever opened the pull request:
Updates for the `scripts/vendor-boringssl.h` script that make it a bit easier to use and ensure you don't accidentally get an update of BoringSSL you don't want.

### Motivation:

I was working on investigating building a Swift SDK for #223 and ran into a couple of things that I saw could be improved. Not all need to be done, but at the very least I feel the boringssl revision lock should be considered. @Lukasa 

### Modifications:

- Add missing invocation of `$sed` in the script- `gsed` was being invoked instead, although everywhere else `$sed` was used (typo).
- Support locking `BORINGSSL_REVISION` to a specific revision number through an argument to the script. This way the script can be re-run without unintentionally upgrading boringssl.
- Remove the non-existent `boringssl_prefix_symbols_nasm.inc` exclude from the `CCryptoBoringSSL` target. I'm guessing it was there for an older version of boringssl, but no longer there with the current revision?

### Result:

The `scripts/vendor_boringssl.sh` script behaves a bit better and can be run without upgrading the boringssl version unless explicitly desired.

Lukasa commented on scripts/vendor-boringssl.sh:
```diff
@@ -101,9 +101,9 @@ function mangle_symbols {
+        docker run -t -i --rm --privileged -v"$(pwd)":/src -w/src --platform linux/arm64 swift:6.0-jammy \
             swift build --product CCryptoBoringSSL
-        docker run -t -i --rm --privileged -v"$(pwd)":/src -w/src --platform linux/amd64 swift:5.10-jammy \
+        docker run -t -i --rm --privileged -v"$(pwd)":/src -w/src --platform linux/amd64 swift:6.0-jammy \
```
We should usually keep this on the oldest supported Swift compiler, which for now remains 5.10. It will become 6.0 when 6.2 goes GA.

Lukasa commented on scripts/vendor-boringssl.sh:
```diff
@@ -170,7 +170,8 @@ echo "CLONING boringssl"
 cd "$SRCROOT"
-BORINGSSL_REVISION=$(git rev-parse HEAD)
+BORINGSSL_REVISION=035e720641f385e82c72b7b0a9e1d89e58cb5ed5
+git checkout $BORINGSSL_REVISION
```
I am inclined to flip this behaviour and say that we should enable the passing of an optional flag to the script that is a revision and, if passed, we should check that out. That way we don't end up encoding the hash in the script itself, but enable optional repeatable runs.

xtremekforever replied:
I can do that!! Anything that allows us to lock the version when just maintaining and updating other pieces of the script.

xtremekforever replied:
So now the revision can be passed to the script like this:

```console
$ ./scripts/vendor-boringssl.sh 035e720641f385e82c72b7b0a9e1d89e58cb5ed5
REMOVING any previously-vendored BoringSSL code
CLONING boringssl
Cloning into '/tmp/.workingN8YfSl/src/boringssl.googlesource.com/boringssl'...
remote: Finding sources: 100% (25/25)
remote: Total 124009 (delta 74986), reused 124005 (delta 74986)
Receiving objects: 100% (124009/124009), 289.25 MiB | 77.32 MiB/s, done.
Resolving deltas: 100% (74986/74986), done.
Updating files: 100% (7939/7939), done.
CHECKING OUT boringssl@035e720641f385e82c72b7b0a9e1d89e58cb5ed5
Note: switching to '035e720641f385e82c72b7b0a9e1d89e58cb5ed5'.
```

As you can see, it will give a message about using that revision before running the `git checkout` command.

xtremekforever commented:
> Looks like shellcheck has a few concerns.

I've updated the script with the quotes as requested. Also noticed that if an argument was not passed the script would give this error:

```
./scripts/vendor-boringssl.sh: line 49: $1: unbound variable
```

And fixed this one as well!

Lukasa commented:
Thanks!
